### PR TITLE
Adding Azure specific architecture + references

### PR DESCRIPTION
**Pull request by codyhoag**
Supersedes #16466. 

/cc @Makdaam @maxwelldb

### CODE REVIEW

**Comment by codyhoag** [comment]
@openshift/team-documentation can someone peer review this, please? Thanks!

File: architecture/additional_concepts/storage.adoc [hunk 1]
```diff
@@ -306,10 +306,22 @@ ifdef::openshift-dedicated,openshift-aro,openshift-online[]
 The following restrictions apply when using PVs with {product-title}:
 endif::[]
 
+ifdef::openshift-aro[]
+[IMPORTANT]
+====
+Persistent volumes are backed by the following storage classes:
+
+* azure-disk
```
Comment (bmcelvee) [inline architecture/additional_concepts/storage.adoc]: Suggestion: s/azure-disk/`azure-disk` (same with azure-file)

File: architecture/additional_concepts/storage.adoc [hunk 2]
```diff
@@ -306,10 +306,22 @@ ifdef::openshift-dedicated,openshift-aro,openshift-online[]
...
+* azure-file
+
+Both of these storage technologies rely on the Azure storage infrastructure. The {product-title} Service does not back up persistent volumes and, therefore, recommends using external Services for storing sensitive data or application state.
```
Comment (bmcelvee) [inline architecture/additional_concepts/storage.adoc]: s/volumes and, therefore,/volumes and

File: architecture/azure/processes.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+After the resource provider switches the cluster into the appropriate
+provisioning state, it performs a back up of etcd. The process continues by
+generating all required configuration and template files and placing them in
+their appropriate storage accounts.
+
+=== Master updates
+
+The master nodes update first. The resource provider checks if the current
+hash of master-000000 matches the desired hash value. If the values match,
```
Comment (bmcelvee) [inline architecture/azure/processes.adoc]: Suggestion: s/master-000000/`master-000000`

File: architecture/azure/processes.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+the master remains unchanged. If the values do not match, the resource provider
+updates the master by draining, stopping, and updating it to the desired VM; the
+master then restarts.
+
+The process repeats with the other two masters in order, so that only a single
+master stops at a time.
+
+=== Infra and Compute node updates
+
+Next the infra nodes update. In normal update flow, the resource provider creates
```
Comment (bmcelvee) [inline architecture/azure/processes.adoc]: I'm not certain on the capitalization of infra here - my [quick] search yielded mixed results.

s/ss-infra-*/`ss-infra-*`

**Comment by bmcelvee** [comment]
Just a few nits, otherwise LGTM!

**Comment by codyhoag** [comment]
Thanks, @bmcelvee !

**Pull request merged**